tensorboard/webapp/metrics/store/metrics_reducers.ts

```diff
@@ -941,8 +941,23 @@ const reducer = createReducer(
     };
   }),
   on(actions.selectTimeEnableToggled, (state) => {
+    let nextCardStepIndexMap = {...state.cardStepIndex};
+    // Updates cardStepIndex only when toggle to enable selectedTime.
+    if (!state.selectTimeEnabled) {
+      const {min} = state.stepMinMax;
+      const startStep = min === Infinity ? 0 : min;
+      nextCardStepIndexMap = generateNextCardStepIndexFromSelectedTime(
+        state.cardStepIndex,
+        state.cardMetadataMap,
+        state.timeSeriesData,
+        // Sets start step to min step when select time is null
+        state.selectedTime ?? {start: {step: startStep}, end: null}
```

Comment:
Should we just set state.selectedTime to be {start: {step: startStep}, end: null} if it is null? Rather than the UI having special case for selectedTime is null then just set it to what the UI is pretending it to be.

const nextSelectedTime = state.selectedTime ?? {start: {step: startStep}, end: null};

And use  nextSelectedTime here and in the thing you are returning on L958.

Reply:
That's interesting idea. I think fundamentally it's not a bad idea to separate the ngrx state and handle what we want to show in UI. 
For example, if user click enable and then toggle again, we then set the selectTime while users are not set it really. It does not feel quite right but I tried to think about caveats but no really strong reasons.

On the other hand, I can also see the convenience of set the selected time here and we do not need to handle cases where select time is null. It is always something when it is enabled. (I can foreseen there are a number of cleanup work, including `timeSelectionCleared` action removal)

What do you think?

for the sake of time I go ahead and made those changes in second commit https://github.com/tensorflow/tensorboard/pull/5641/commits/c75b375d9ad7867abeb09dbd3ed08f14f141cb12

Reply:
Thinking about it a bit more, there is one caveat: What if the "min" changes over time as new runs are loaded? Under the old style (where selectedTime is null) then all the UI would automatically adjust to this new min when linked time is (re)enabled. Now, with the new code, it will be stuck on the old min.

However, I think it's such a small caveat I wouldn't worry about spending time or code on it. Any defaults we choose when user turns on linked time is very likely wrong anyways.